src/compiler/typescript_to_rust/azle_generate/src/canister_method/errors/mismatched_post_deploy_params.rs

```diff
@@ -0,0 +1,220 @@
+use annotate_snippets::{
+    display_list::{DisplayList, FormatOptions},
+    snippet::{Annotation, AnnotationType, Slice, Snippet, SourceAnnotation},
+};
+use swc_common::{source_map::Pos, BytePos, Span};
+
+use crate::{
+    canister_method::AnnotatedFnDecl,
+    traits::GetSourceFileInfo,
+    ts_ast::{source_map::Range, SourceMapped},
+};
+
+pub type SmAnFnDecl<'a> = SourceMapped<'a, AnnotatedFnDecl>;
+
+/// Returned when Azle detects a difference between the params in $init and $postUpgrade
+///
+/// # Example
+///
+/// ```ts
+/// import { $init, $postUpgrade, int32 } from 'azle';
+///
+/// $init;
+/// export function init(p1: boolean, p2: string, p3: int32): void {
+///     console.log('Initialization complete.');
+/// }
+///
+/// $postUpgrade;
+/// export function postUpgrade(p1: int32, p2: boolean): void {
+///     console.log('PostUpgrade complete.');
+/// }
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MismatchedPostDeployParams {
+    pub origin: String,
+    pub line_number: usize,
+    pub source: String,
+    pub ranges: Vec<(usize, usize)>,
+}
+
+impl MismatchedPostDeployParams {
```

Comment:
We need this to work if init and post_upgrade are in separate files

Reply:
Done by making the error message very simple instead. See c1aea753d74f473388ac09c48f1a244421af856a